integr8ly/integreatly-operator
# INTLY-6522 alerts for  cro resource state metrics

# Description

Add alerts on the phase status returned by the cloud resource operator
https://issues.redhat.com/browse/INTLY-6522
follow on from https://issues.redhat.com/browse/INTLY-6521 where the metrics were exposed.

## Type of change



 

## Checklist
- [x] This change requires a documentation update  SOP url and doc need to be update , future jira for this work https://issues.redhat.com/browse/INTLY-7756

### Verification
BYOC cluster required for verification
- clone this branch
- use this cro image quay.io/austincunningham/cloud-resource-operator:0.15.2 to expose the metrics needed
- install the latest version of the clould-resource-operator in rhmi
```bash
go get github.com/integr8ly/cloud-resource-operator@master
```
- `make cluster/prepare/project`
- `make cluster/prepare/crd`
- change `UseClusterStorage` to false in the rhmi cr
- deploy the cr `oc create -f deploy/crds/examples/rhmi.cr.yaml`
- `make code/run`
- check prometheus for new alerts
    - [productname]-PostgresResourceStatusPhaseFailed
    - [productname]-PostgresResourceStatusPhasePending
e.g.
![image](https://user-images.githubusercontent.com/16667688/82548466-3ad74a00-9b53-11ea-8a89-21d19d731f79.png)

    - [productname]-RedisResourceStatusPhaseFailed
    - [productname]-RedisResourceStatusPhasePending
![image](https://user-images.githubusercontent.com/16667688/82551919-167e6c00-9b59-11ea-93a0-aebd6e378317.png)
